test/ServerComparison.FunctionalTests/ResponseCompressionTests.cs

```diff
@@ -30,210 +28,80 @@ public ResponseCompressionTests(ITestOutputHelper output) : base(output)
-        {
-            return ResponseCompression(ServerType.IISExpress, 
-                runtimeFlavor, 
-                RuntimeArchitecture.x64, 
-                CheckAppCompressionAsync, 
-                applicationType,
-                hostCompression: true, 
-                hostingModel: hostingModel, 
-                ancmVersion: ancmVersion);        
-        }
-
-        // WebListener
-        [ConditionalTheory]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
-        [InlineData(RuntimeFlavor.Clr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone)]
-        public Task ResponseCompression_WebListener_NoCompression(RuntimeFlavor runtimeFlavor, ApplicationType applicationType)
+        [Theory]
+        [MemberData(nameof(NoCompressionTestVariants))]
+        public Task ResponseCompression_NoCompression(TestVariant variant)
         {
-            return ResponseCompression(ServerType.WebListener, runtimeFlavor, RuntimeArchitecture.x64, CheckNoCompressionAsync, applicationType, hostCompression: false);
+            return ResponseCompression(variant, CheckNoCompressionAsync, hostCompression: false);
         }
 
-        // WebListener doesn't support HostCompression
-        // "The archive entry was compressed using an unsupported compression method."
-
-        [ConditionalTheory]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
-        [InlineData(RuntimeFlavor.Clr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone)]
-        public Task ResponseCompression_WebListener_AppCompression(RuntimeFlavor runtimeFlavor, ApplicationType applicationType)
-        {
-            return ResponseCompression(ServerType.WebListener, runtimeFlavor, RuntimeArchitecture.x64, CheckAppCompressionAsync, applicationType, hostCompression: false);
-        }
-        
-        [ConditionalTheory]
-        [OSSkipCondition(OperatingSystems.Linux | OperatingSystems.MacOSX)]
-        [InlineData(RuntimeFlavor.Clr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Portable)]
-        [InlineData(RuntimeFlavor.CoreClr, ApplicationType.Standalone)]
-        public Task ResponseCompression_WebListener_AppAndHostCompression(RuntimeFlavor runtimeFlavor, ApplicationType applicationType)
-        {
-            return ResponseCompression(ServerType.WebListener, runtimeFlavor, RuntimeArchitecture.x64, CheckAppCompressionAsync, applicationType, hostCompression: true);
-        }
+        public static TestList HostCompressionTestVariants
+            => TestMatrix.ForServers(ServerType.IISExpress, ServerType.Nginx)
+                .WithTfms(Tfm.NetCoreApp22, Tfm.Net461)
```

Comment:
Do we want a `WithLatestTfms()` and `WithLatestCore`/`WithLatestFullFx` helpers?